File: Diagrammatize.Design.Pattern/code/python/Bridge/A3/IncreaseDisplay.py
from CountDisplay import CountDisplay

class IncreaseDisplay(CountDisplay):
    def __init__(self, impl, step):
        super().__init__(impl)
        self.step = step

    def increaseDisplay(self, level):
        count = 0
        for i in range(level):
            self.multiDisplay(count)
            count += self.step

File: Diagrammatize.Design.Patterns/code/cxx/Bridge/A3/IncreaseDisplay.cpp
#include "IncreaseDisplay.hpp"

IncreaseDisplay::IncreaseDisplay(DisplayImpl* impl, int step_): CountDisplay(impl), step(step_) {
}

void IncreaseDisplay::increaseDisplay(int level) {
    int count = 0;
    for (int i = 0; i < level; i++) {
        multiDisplay(count);
        count += step;
    }
}

